src/models/archaeologist.ts

```diff
@@ -66,22 +67,15 @@ export class Archaeologist {
 
   async setupCommunicationStreams() {
     await this._setupSarcophagusNegotiationStream();
+    await this._setupPublicKeyStream();
   }
 
   async initNode(arg: {
     config: PublicEnvConfig;
     web3Interface: Web3Interface;
     idFilePath?: string;
   }) {
-    this.node = await this.createLibp2pNode(arg.idFilePath);
-    this.envConfig = arg.config;
-    this.web3Interface = arg.web3Interface;
-
-    return this.node;
-  }
-
-  async createLibp2pNode(idFilePath?: string): Promise<Libp2p> {
```

Comment:
No reason that I could see to keep this `createLibp2pNode` separate from `initNode`. Probably leftovers from previous iterations of the codebase. So I've merged them so it's easier to find "the code that does arch node initialisation"